### ISSUE TITLE
Add json validation when ingesting jsonl files for training

### ISSUE THREAD

**Issue by suchenzang** [facebookresearch/metaseq#139]
Can be some simple bash like:
```
ls *.jsonl > file_names.txt
while read f; do
  if cat $f | jq -e . >/dev/null 2>&1; then
    echo "Parsed JSON successfully: $f"
  else
    echo "Failed to parse JSON: $f"
  fi
done <file_names.txt
```
in the data dirs.

Goal is just to check that json files aren't truncated somehow.  Can also add as a step when building index file for JsonlDatasets.

**Comment by KUNAL1612** [comment]
@suchenzang So to confirm, the json validation here is just making sure we can read a jsonl file and has nothing to do with the actual content of the files?

**Comment by punitkoura** [comment]
> @suchenzang So to confirm, the json validation here is just making sure we can read a jsonl file and has nothing to do with the actual content of the files?

I think we also want to ensure that each json line is properly formatted.

**Comment by suchenzang** [comment]
Yup, seconding @punitkoura : this task came up because we found corrupted json files during one of our experiment runs (truncated line).  A simple validation here to check that we are including valid json when loading json files will help guard against this error in the future.

**Comment by KUNAL1612** [comment]
Fixes in #212

**Comment by punitkoura** [comment]
Fixed by @KUNAL1612 , closing
